massa-bootstrap/src/server/allow_block_list.rs

```diff
@@ -0,0 +1,114 @@
+        let (allow_list, block_list) =
+            AllowBlockListInner::load_allow_block_lists(&allow_path, &block_path)?;
+        Ok(Self {
+            inner: Arc::new(RwLock::new(AllowBlockListInner {
+                allow_list,
+                block_list,
+            })),
+            allow_path: Cow::from(allow_path),
+            block_path: Cow::from(block_path),
+        })
+    }
+
+    /// Checks if the allow/block list is up to date with a read-lock
+    /// Creates a new list, and replaces the old one in a write-lock
+    pub(crate) fn update(&mut self) -> Result<(), String> {
+        let read_lock = self.inner.read();
+        let (new_allow, new_block) =
+            AllowBlockListInner::load_allow_block_lists(&self.allow_path, &self.block_path)?;
+        let allow_delta = new_allow != read_lock.allow_list;
+        let block_delta = new_block != read_lock.block_list;
+        if allow_delta || block_delta {
+            // Ideally this scope would be atomic
+            let mut mut_inner = {
+                drop(read_lock);
+                self.inner.write()
+            };
+
+            if allow_delta {
+                mut_inner.allow_list = new_allow;
+            }
+            if block_delta {
+                mut_inner.block_list = new_block;
+            }
+        }
+        Ok(())
+    }
+
+    #[cfg_attr(test, allow(unreachable_code, unused_variables))]
+    pub(crate) fn is_ip_allowed(&self, remote_addr: &SocketAddr) -> Result<(), String> {
+        #[cfg(test)]
+        return Ok(());
+
+        let ip = normalize_ip(remote_addr.ip());
+        // whether the peer IP address is blacklisted
+        let read = self.inner.read();
+        if let Some(ip_list) = &read.block_list && ip_list.contains(&ip) {
+            massa_trace!("bootstrap.lib.run.select.accept.refuse_blacklisted", {"remote_addr": remote_addr});
+            Err(format!("IP {} is blacklisted", &ip))
+            // whether the peer IP address is not present in the whitelist
+        } else if let Some(ip_list) = &read.allow_list && !ip_list.contains(&ip){
+            massa_trace!("bootstrap.lib.run.select.accept.refuse_not_whitelisted", {"remote_addr": remote_addr});
+            Err(format!("A whitelist exists and the IP {} is not whitelisted", &ip))
+        } else {
+            Ok(())
+        }
+    }
+}
+
+impl AllowBlockListInner {
+    #[allow(clippy::result_large_err)]
+    #[allow(clippy::type_complexity)]
+    fn load_allow_block_lists(
+        allowlist_path: &Path,
+        blocklist_path: &Path,
+    ) -> Result<(Option<HashSet<IpAddr>>, Option<HashSet<IpAddr>>), String> {
+        let allow_list = Self::load_list(allowlist_path)?;
+        let block_list = Self::load_list(blocklist_path)?;
+        Ok((allow_list, block_list))
+    }
+
+    fn load_list(list_path: &Path) -> Result<Option<HashSet<IpAddr>>, String> {
+        let Ok(list) = std::fs::read_to_string(list_path) else {
```

Comment:
Print an error when std::fs::read_to_string fail

Reply:
done, though instead of logging/printing, it propogates the error up (I'll log the errors where the errors are eventually captured)

Reply:
edit: propogating the error up is not what we want.

I'll add a debug! log

Reply:
personally if i provide a file path, i want the software load it correctly, if he fail i want to know it. 
So i think error level for the log is better here. Maybe not ?

Reply:
You're right. debug! is definately not the right one.

What do you think about `warn!`? In this context, the intent to have a white/black list of IPs is not available, and aside from that, everything is a-okay.

Reply:
i don't want everything is ok if i provide a file and he failed to load my whitelist/blacklist. I think the end user must not miss this information.

maybe @AurelienFT would have an opinion too

Personally i can `panic!` here :boom:

Reply:
The precedent behaviour was : if there is no whitelist than everyone is allowed. If there is an existing whitelist but empty everyone is forbidden

So in the case of the whilelist no being here isn't an issue. This is our way to have "everyone allow" and "nobody allow" with this file.

Reply:
Not exactly sure what the ideal resolution for this is, but I'm happy to `panic!` here as the panic would be triggered on startup.

Would be problematic if in the future we would like to change the file while things are running, or if you change the name of the file in the file-system.

Reply:
Don't change the behavior in this PR (for this month). You can just add a warning.

Reply:
From what I can work out, the testnet 20 behavior was that the bootstrap server would die silently and independantly of the rest of the system by `?` returning errors, ending up in the task-handler. The error is captured in a `let _ = self.join_handle.await?;`, and the returning Err is handled with `.expect("bootstrap server shutdown failed")`.

The behavior here should be the same, with some extra `error!` and `info!` tracing macros thrown in for good measure.